src/modules/mc_pos_control/PositionControl/PositionControl.cpp

```diff
@@ -63,6 +63,12 @@ void PositionControl::setThrustLimits(const float min, const float max)
 	_lim_thr_max = max;
 }
 
+void PositionControl::updateHoverThrust(const float hover_thrust_new)
+{
+	_vel_int(2) += hover_thrust_new - _hover_thrust;
```

Comment:
One thing: The integrator gets changed and applied before getting limited
```
// limit thrust integral
_vel_int(2) = math::min(fabsf(_vel_int(2)), _lim_thr_max) * math::sign(_vel_int(2));
```
so it could be over the limit with a big hover thrust change and then get limited in the next loop iteration... Usually, that should not happen since we try to unload the integrator.